src/sidebar/components/annotation-omega.js

```diff
@@ -17,25 +19,34 @@ function AnnotationOmega({
   replyCount,
   showDocumentInfo,
 }) {
+  // An annotation will have a draft if it is being edited
   const draft = useStore(store => store.getDraft(annotation));
+  // An annotation will always have a "state": either its current draft or
+  // state derived from the annotation itself
+  const annotationState = useStore(store => {
+    return store.getAnnotationState(annotation);
+  });
+
+  const isSaving = false;
+  const isEditing = !!draft && !isSaving;
+
+  const createDraftFn = useStore(store => store.createDraft);
 
   // Annotation metadata
   const hasQuote = !!quote(annotation);
 
-  // Local component state
-  // TODO: `isEditing` will also take into account `isSaving`
-  const isEditing = !!draft;
+  // Create a new (updated) draft based on the existing state extended with
+  // any changes in `newState`
+  const updateState = newState => {
+    createDraftFn(annotation, Object.assign({}, annotationState, newState));
+  };
 
-  const annotationState = () =>
-    draft || {
-      text: annotation.text,
-      tags: annotation.tags,
-      annotation,
-    };
+  const onEditTags = changes => {
+    updateState({ tags: changes.tags });
+  };
 
-  // TODO
-  const fakeOnEditText = () => {
-    alert('TBD');
+  const onEditText = changes => {
+    updateState({ text: changes.text });
   };
```

Comment:
Hi Lyza,

I would caution against trying to port the `state` property of the old annotation controller too directly into new code. Amongst other things it was a bad (overly generic) choice of name on my part. It isn't really a piece of mutable state but rather the result of choosing the text/tags/privacy from either the draft or the annotation depending on whether it is being edited or not.

The simplest way to map this into idiomatic React IMO would be to have conditionally assigned variables, along the lines of:

```diff
diff --git a/src/sidebar/components/annotation-omega.js b/src/sidebar/components/annotation-omega.js
index 2b0aa40d9..51b77176e 100644
--- a/src/sidebar/components/annotation-omega.js
+++ b/src/sidebar/components/annotation-omega.js
@@ -21,32 +21,21 @@ function AnnotationOmega({
 }) {
   // An annotation will have a draft if it is being edited
   const draft = useStore(store => store.getDraft(annotation));
-  // An annotation will always have a "state": either its current draft or
-  // state derived from the annotation itself
-  const annotationState = useStore(store => {
-    return store.getAnnotationState(annotation);
-  });
+  const text = draft ? draft.text : annotation.text;
+  const tags = draft ? draft.tags : annotation.tags;
 
   const isSaving = false;
   const isEditing = !!draft && !isSaving;
-
-  const createDraftFn = useStore(store => store.createDraft);
-
-  // Annotation metadata
   const hasQuote = !!quote(annotation);
 
-  // Create a new (updated) draft based on the existing state extended with
-  // any changes in `newState`
-  const updateState = newState => {
-    createDraftFn(annotation, Object.assign({}, annotationState, newState));
-  };
+  const createDraft = useStore(store => store.createDraft);
 
-  const onEditTags = changes => {
-    updateState({ tags: changes.tags });
+  const onEditTags = ({ tags }) => {
+    createDraft(annotation, { ...draft, tags });
   };
 
-  const onEditText = changes => {
-    updateState({ text: changes.text });
+  const onEditText = ({ text }) => {
+    createDraft(annotation, { ...draft, text });
   };
 
   return (
@@ -63,13 +52,13 @@ function AnnotationOmega({
         annotation={annotation}
         isEditing={isEditing}
         onEditText={onEditText}
-        text={annotationState.text}
+        text={text}
       />
       {isEditing && (
-        <TagEditor onEditTags={onEditTags} tagList={annotationState.tags} />
+        <TagEditor onEditTags={onEditTags} tagList={tags} />
       )}
       {!isEditing && (
-        <TagList annotation={annotation} tags={annotationState.tags} />
+        <TagList annotation={annotation} tags={tags} />
       )}
     </div>
   );
```

There may be a case in future where other parts of the app need to see the combined annotation + draft changes, in which case the approach you took here of providing a selector in the store that returns that would be quite reasonable. Until then I'd be inclined to view this as a UI concern which doesn't belong there.